django-stubs/contrib/auth/models.pyi

```diff
@@ -38,34 +49,32 @@ class Group(models.Model):
     permissions = models.ManyToManyField(Permission)
     def natural_key(self) -> tuple[str]: ...
 
-_T = TypeVar("_T", bound=Model)
-
-class UserManager(BaseUserManager[_T]):
+class UserManager(BaseUserManager[_UserType]):
     def create_user(
         self, username: str, email: str | None = ..., password: str | None = ..., **extra_fields: Any
-    ) -> _T: ...
+    ) -> _UserType: ...
     def create_superuser(
         self, username: str, email: str | None = ..., password: str | None = ..., **extra_fields: Any
-    ) -> _T: ...
+    ) -> _UserType: ...
     def with_perm(
         self,
         perm: str | Permission,
         is_active: bool = ...,
         include_superusers: bool = ...,
         backend: str | None = ...,
         obj: Model | None = ...,
-    ) -> QuerySet[_T]: ...
+    ) -> QuerySet[_UserType]: ...
 
 class PermissionsMixin(models.Model):
     is_superuser = models.BooleanField()
     groups = models.ManyToManyField(Group)
     user_permissions = models.ManyToManyField(Permission)
 
-    def get_user_permissions(self, obj: _AnyUser | None = ...) -> set[str]: ...
-    def get_group_permissions(self, obj: _AnyUser | None = ...) -> set[str]: ...
-    def get_all_permissions(self, obj: _AnyUser | None = ...) -> set[str]: ...
-    def has_perm(self, perm: str, obj: _AnyUser | None = ...) -> bool: ...
-    def has_perms(self, perm_list: Iterable[str], obj: _AnyUser | None = ...) -> bool: ...
+    def get_user_permissions(self, obj: AbstractUser | None = ...) -> set[str]: ...
```

Comment:
I'm not sure if there's enough tests against how this one is used, but this may want to stay wide as `_AnyUser` :thinking: